File: JavaScript Implementation/symbol-table/VariableSymbol.js
const Symbol = require('./Symbol');

class VariableSymbol extends Symbol {
  constructor(name, type, isConstant = false, value=null) {
    super(name,type);
    if(value != null) 
      this.value = value
    else if(type.name === "INTEGER")
      this.value = 0
    else if(type.name === "STRING" || type.name === "CHAR")
      this.value = ''
    else if(type.name === "BOOLEAN")
      this.value = false
    else if(type.name === "FLOAT" || type.name === "DOUBLE") 
      this.value = 0.0 
    this.isConstant = isConstant;
    }

    setValue(value) {
      this.value = value;
    }
    
    toString() {
      return `VariableSymbol(${this.getName()}, ${this.getType()},${this.value})`;
    }

    static create(name, type, isConstant, value) {
      return new this(name, type, isConstant, value);
    }
  }
  
  module.exports = VariableSymbol;

File: Java Implementation/src/symbol/VariableSymbol.java
package symbol;

public class VariableSymbol extends Symbol {

    private boolean isConstant = false;
    private Object value = null;

    public VariableSymbol(String name, String type, Boolean isConstant, Object value) {
        super(name, type);

        this.isConstant = isConstant;
    }

    public String toString() {
        return this.getName() + " " + this.getType() + " " + this.value;
    }
}
